### PR TITLE
Add spellcheck to CI

### PR DESCRIPTION
**Pull request by notgull**
As [above](https://github.com/rust-windowing/winit/pull/3557), so below.

### CODE REVIEW

**Comment by notgull** [comment]
CI error is fixed in https://github.com/rust-windowing/raw-window-handle/pull/162

**Comment by notgull** [comment]
Changed to `typos` as per the decision in https://github.com/rust-windowing/winit/pull/3565

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,7 +40,7 @@
 
 ## 0.4.1 (2021-11-19)
 
-* Added an impl of `HasRawWindowHandle` for `&T`, `Rc<T>`, and `Arc<T>`. The impls for `Rc<T>` and `Arc<T>` require the `alloc` feature.
+* Added an impl of `HasRawWindowHandle` for `&T`, `Rc<T>`, and `Arc<T>`. The implementations for `Rc<T>` and `Arc<T>` require the `alloc` feature.
```
Comment (Lokathor) [inline CHANGELOG.md:43]: I still believe that "impls" is correct plural, because "impl" is the singular ;3

File: src/lib.rs [hunk 2]
```diff
@@ -159,7 +159,7 @@ pub enum RawWindowHandle {
     ///
     /// ## Availability Hints
     /// This variant is present regardless of windowing backend and likely to be used with
-    /// EGL_MESA_platfrom_gbm or EGL_KHR_platfrom_gbm.
+    /// EGL_MESA_platform_gbm or EGL_KHR_platform_gbm.
```
Comment (Lokathor) [inline src/lib.rs:162]: Is the diff being weird? This looks like text was replaced with identical text

Comment (notgull) [reply src/lib.rs:162]: "plat**from**" is being replaced with "plat**form**"

**Review by Lokathor — APPROVED** [review]

**Pull request merged**